Data/ByteString.hs

```diff
@@ -1527,6 +1530,17 @@ stripSuffix bs1@(BS _ l1) bs2@(BS _ l2)
 isInfixOf :: ByteString -> ByteString -> Bool
 isInfixOf p s = null p || not (null $ snd $ breakSubstring p s)
 
+-- | /O(n)/ Check whether a 'ByteString' represents valid UTF-8.
+--
+-- @since 0.11.2.0
+isValidUtf8 :: ByteString -> Bool
+isValidUtf8 (BS ptr len) = accursedUnutterablePerformIO $ unsafeWithForeignPtr ptr $ \p -> do 
+  CInt i <- cIsValidUtf8 p (CSize . fromIntegral $ len)
+  pure $ i /= 0
```

Comment:
Slightly shorter:

```suggestion
  i <- cIsValidUtf8 p (fromIntegral len)
  pure $ i /= 0
```

Alternatively, more explicit about the integer conversion:

```suggestion
    i <- cIsValidUtf8 p (intToCSize len)
    pure $ i /= 0
  where
    intToCSize :: Int -> CSize
    intToCSize = fromIntegral
```